zcash_client_backend/src/welding_rig.rs

```diff
@@ -152,8 +154,9 @@ pub fn scan_block<P: consensus::Parameters>(
                     .map(|output| &mut output.witness)
                     .collect();
 
-                if let Some(output) = scan_output::<P>(
-                    block.height as u32,
+                if let Some(output) = scan_output(
+                    params,
+                    block.height.into(),
```

Comment:
```suggestion
                    block.height(),
```

Reply:
This change results in:
```
159 |                     block.height(),
    |                     ^^^^^ value borrowed here after partial move
```
How do I need to change `height()` so that this doesn't happen?

Reply:
That error indicates the issue is further up. Somewhere above this point, a value is moved from inside `block`; this poisons `block` such that we can't take any reference to it (because "it" doesn't exist any longer - we just have a bag of its innards we can refer to individually).

Reply:
The move occurs from ```block.vtx.into_iter()``` - what's the best practice for how to write this?

Reply:
Just separate `let` bindings for each element that will eventually be used?

Reply:
Yeah, the separate `block_height` binding you added works well here.